notifications_utils/recipients.py

```diff
@@ -177,15 +187,20 @@ def get_rows(self):
                     insert_or_append_to_dict(output_dict, key, None)
 
             if index < self.max_rows:
-                yield Row(
+                row = Row(
                     output_dict,
                     index=index,
                     error_fn=self._get_error_for_field,
                     recipient_column_headers=self.recipient_column_headers,
                     placeholders=self.placeholders_as_column_keys,
                     template=self.template,
                 )
+                if rows_as_a_list_is_empty:
+                    self.rows_as_list.append(row)
```

Comment:
You don’t need to build up the list here (and there’s a risk that you’ll end up building a partial list).

If this method doesn’t build up the list then the API can use it as a pure generator, which should reduce memory usage.